### ISSUE TITLE
DispatchConsumerConcurrency might be misplaced on the connection factory

### ISSUE THREAD

**Issue by danielmarbach** [rabbitmq/rabbitmq-dotnet-client#1668]
### Is your feature request related to a problem? Please describe.

Can also be considered v7 feedback.

Generally, we have one connection factory per endpoint. For every receive pump we create a channel from that connection. Every pump might have its own concurrency control associated with it, and ideally we would like to offload the concurrency control to the client.

Unfortunately, as of today, you can only set the `DispatchConsumerConcurrency` on the connection factory. The setting then gets copied into the `ConnectionConfig` which is passed to the `ChannelBase`. 

That relationship seems a little awkward because ideally, you would want to control the dispatcher concurrency per channel. Given that we recently renamed the model creation methods, I think it would make sense to move the dispatcher concurrency to the `CreateChannel` method instead of having it on the connection factory.

With today's approach, we have to do an ugly workaround like locking the connection factory, override the dispatcher concurrency, create a model and unlock the connection factory.

### Describe the solution you'd like

See above

**Comment by danielmarbach** [comment]
I haved added a draft PR https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1669

**Comment by danielmarbach** [comment]
FYI Currently we have this code in place

```csharp
            try
            {
                await lockObject.WaitAsync(cancellationToken).ConfigureAwait(false);

                connectionFactory.AutomaticRecoveryEnabled = automaticRecoveryEnabled;
                connectionFactory.ConsumerDispatchConcurrency = consumerDispatchConcurrency;

                var connection = await connectionFactory.CreateConnectionAsync(endpoints, connectionName, cancellationToken).ConfigureAwait(false);

                connection.ConnectionBlocked += (sender, e) => Logger.WarnFormat("'{0}' connection blocked: {1}", connectionName, e.Reason);
                connection.ConnectionUnblocked += (sender, e) => Logger.WarnFormat("'{0}' connection unblocked}", connectionName);

                connection.ConnectionShutdown += (sender, e) =>
                {
                    if (e.Initiator == ShutdownInitiator.Application && e.ReplyCode == 200)
                    {
                        return;
                    }

                    Logger.WarnFormat("'{0}' connection shutdown: {1}", connectionName, e);
                };

                return connection;
            }
            finally
            {
                lockObject.Release();
            }
```

Another option might be move the setting to the connection level but right now it seems the relationship between Channels and ConsumerDispatcher that controls the concurrency is 1:1

**Comment by lukebakken** [comment]
Fixed by #1669 and #1671! Thank you @danielmarbach

**Comment by danielmarbach** [comment]
You are welcome. Glad this is solved

I was also thinking about opening the discussion of whether it really makes sense to have certain settings like recovery enabled or the concurrency on the connection factory or whether those should be moved to the create connection call instead but since those things are not that relevant for us I don't want to further add stuff and potentially delay things
